src/cdk/tree/tree.ts

```diff
@@ -311,6 +571,513 @@ export class CdkTree<T, K = T> implements AfterContentChecked, CollectionViewer,
       CdkTreeNode.mostRecentTreeNode.data = nodeData;
     }
   }
+
+  /** Whether the data node is expanded or collapsed. Returns true if it's expanded. */
+  isExpanded(dataNode: T): boolean {
+    return (
+      this.treeControl?.isExpanded(dataNode) ??
+      this._expansionModel?.isSelected(this._getExpansionKey(dataNode)) ??
+      false
+    );
```

Comment:
```suggestion
    return !!(
      this.treeControl?.isExpanded(dataNode) ||
      this._expansionModel?.isSelected(this._getExpansionKey(dataNode)));
```
optional: I find the logical `||` with boolean coercion more readable in this situation